classes/citation/Citation.php

```diff
@@ -20,99 +16,70 @@
 
 namespace PKP\citation;
 
-class Citation extends \PKP\core\DataObject
-{
-    /**
-     * Constructor.
-     *
-     * @param string $rawCitation an unparsed citation string
-     */
-    public function __construct($rawCitation = null)
-    {
-        parent::__construct();
-        $this->setRawCitation($rawCitation);
-    }
-
-    //
-    // Getters and Setters
-    //
-
-    /**
-     * Replace URLs through HTML links, if the citation does not already contain HTML links
-     *
-     * @return string
-     */
-    public function getCitationWithLinks()
-    {
-        $citation = $this->getRawCitation();
-        if (stripos($citation, '<a href=') === false) {
-            $citation = preg_replace_callback(
-                '#(http|https|ftp)://[\d\w\.-]+\.[\w\.]{2,6}[^\s\]\[\<\>]*/?#',
-                function ($matches) {
-                    $trailingDot = in_array($char = substr($matches[0], -1), ['.', ',']);
-                    $url = rtrim($matches[0], '.,');
-                    return "<a href=\"{$url}\">{$url}</a>" . ($trailingDot ? $char : '');
-                },
-                $citation
-            );
-        }
-        return $citation;
-    }
+use PKP\core\DataObject;
 
+class Citation extends DataObject
+{
     /**
-     * Get the rawCitation
-     *
-     * @return string
+     * Get the rawCitation.
      */
-    public function getRawCitation()
+    public function getRawCitation(): string|null
     {
         return $this->getData('rawCitation');
     }
 
     /**
-     * Set the rawCitation
+     * Set the rawCitation.
      */
-    public function setRawCitation(?string $rawCitation)
+    public function setRawCitation(?string $rawCitation): void
     {
-        $rawCitation = $this->_cleanCitationString($rawCitation ?? '');
+        $rawCitation = $this->cleanCitationString($rawCitation ?? '');
         $this->setData('rawCitation', $rawCitation);
     }
 
     /**
      * Get the sequence number
-     *
-     * @return int
      */
-    public function getSequence()
+    public function getSequence(): int
     {
         return $this->getData('seq');
     }
 
     /**
      * Set the sequence number
-     *
-     * @param int $seq
      */
-    public function setSequence($seq)
+    public function setSequence(int $seq): void
     {
         $this->setData('seq', $seq);
     }
 
-    //
-    // Private methods
-    //
+    /**
+     * Replace URLs through HTML links, if the citation does not already contain HTML links.
+     */
+    public function getRawCitationWithLinks(): string|null
```

Comment:
The same here, if rawCitation always exist, this function will always return string